### PR TITLE
Adds topic name to error msg when create_topic fails

### PR DESCRIPTION
**Pull request by Voldivh**
This PR addresses the issue #409. As the title states, it adds the topic name to the error message when create_topic fails.

Signed-off-by: Voldivh <eloyabmfcv@gmail.com>

### CODE REVIEW

**Review by tfoote — APPROVED** [review]
I was thinking it might make sense to only have one error_msg and just set it within the scope of the if statement. Then call RMW_SET_ERROR_MSG at the end. But that will hide from the macro the line number from which each instance of the error was called. 

lgtm

File: rmw_cyclonedds_cpp/src/rmw_node.cpp [hunk 1]
```diff
@@ -1787,23 +1787,24 @@ static dds_entity_t create_topic(dds_entity_t pp, const char * name, struct ddsi
   return tp;
 }
 
-void set_error_message_from_create_topic(dds_entity_t topic)
+void set_error_message_from_create_topic(dds_entity_t topic, std::string topic_name)
```
Comment (francocipollone) [inline rmw_cyclonedds_cpp/src/rmw_node.cpp:1790]: nit: What about `const std::string& topic_name`?

File: rmw_cyclonedds_cpp/src/rmw_node.cpp [hunk 2]
```diff
@@ -1787,23 +1787,24 @@ static dds_entity_t create_topic(dds_entity_t pp, const char * name, struct ddsi
...
 {
   assert(topic < 0);
   if (DDS_RETCODE_BAD_PARAMETER == topic) {
-    RMW_SET_ERROR_MSG(
-      "failed to create topic because the function was given"
-      " invalid parameters");
+    std::string error_msg = "failed to create topic [" + topic_name +
```
Comment (francocipollone) [inline rmw_cyclonedds_cpp/src/rmw_node.cpp:1794]: nit: Could it be const?

Comment (francocipollone) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp:1794]: Here and there

**Review by francocipollone — COMMENTED** [review]
> But that will hide from the macro the line number from which each instance of the error was called.

+1

File: rmw_cyclonedds_cpp/src/rmw_node.cpp (hunk 2 above)
Comment (Voldivh) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp]: Done.

File: rmw_cyclonedds_cpp/src/rmw_node.cpp (hunk 1 above)
Comment (Voldivh) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp]: Nice catch, thanks!

**Comment by tfoote** [comment]
Gist: https://github.com/ros2/ros2/blob/rolling/ros2.repos

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17301/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11840/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17788/)

**Comment by tfoote** [comment]
@mergify backport humble galactic foxy

**Pull request merged**